packages/block-library/src/comment-template/index.php

```diff
@@ -5,6 +5,84 @@
  * @package WordPress
  */
 
+/**
+ * Convert the flat array of comments to a nested array
+ * containing comments with a `comment_children` property.
+ *
+ * @param WP_Comment[] $comments  - Array of comments.
+ * @return array
+ */
+function convert_to_tree( $comments ) {
```

Comment:
It would be nice to cover this function with a unit test. Example test file for a block:

https://github.com/WordPress/gutenberg/blob/trunk/phpunit/class-block-library-navigation-link-test.php

In this case, we can test this method in isolation to make it simpler.

Reply:
Dang it, I've just found out that there is a [`get_children()`](https://developer.wordpress.org/reference/classes/wp_comment/get_children/) method of `WP_Comment` so we probably don't need to the `convert_to_tree` function at all.

Reply:
That would simplify the implementation a bit. Now, that I think about it, there should be an existing API in WP core for most of the processing for comments. The challenge is to discover them 😃

Reply:
ok, done in 09daa22